lib/ldap_fluff/ad_member_service.rb

```diff
@@ -27,13 +27,11 @@ def _groups_from_ldap_data(payload)
   end
 
   # recursively loop over the parent list
-  def _walk_group_ancestry(gids = [])
+  def _walk_group_ancestry(group_dns = [])
     set = []
-    gids.each do |g|
-      filter = group_filter(g) & class_filter
-      search = @ldap.search(:filter => filter, :base => @group_base)
-      if !search.nil? && !search.first.nil?
-        group = search.first
+    group_dns.each do |group_dn|
+      group = @ldap.search(:base => group_dn, :scope => Net::LDAP::SearchScope_BaseObject)
+      if !group.nil?
         set  += get_groups(group[:memberof])
```

Comment:
I think `group` is actually an array, so you might need to use `.first` or similar (see the original code which did `group = search.first`):

<pre>
2.0.0-p353 :016 > ldap.search(:base => 'ou=chemists,dc=example,dc=com', :scope => Net::LDAP::SearchScope_BaseObject)
2015-07-21T15:39:42 [ldap] [D]   op search (290.5ms)  [ filter=, base=ou=chemists,dc=example,dc=com ]
 => [#<Net::LDAP::Entry:0x0000000b1b00e8 @myhash={:dn=>["ou=chemists,dc=example,dc=com"], :ou=>["chemists"], ....]
</pre>